### PR TITLE
Test into qgis docker

### PR DESCRIPTION
**Pull request by marioba**
The tests are now split into:
- unit tests: use the qgis.testing module
- functional tests: run on the official qgis docker whithout the need of external services
- integration tests: run on the official qgis docker and need a running oq_engine

The Travis configuration file is set to run the tests using the scripts in the  `scripts/` directory.

The Travis file must be further fixed to run the tests in different environments and avoid having to delete the docker container between one test and another.

### CODE REVIEW

**Comment by marioba** [comment]
Travis currently fails because one of the integration tests is not working.

File: .travis.yml [hunk 1]
```diff
@@ -23,19 +18,22 @@ install:
...
     - oq webui start 0.0.0.0:8800 --skip-browser &> webui.log &
 
 script:
-    - docker run -ti -e OQ_ENGINE_HOST='http://172.17.0.1:8800' --rm -v $TRAVIS_BUILD_DIR:/io openquake/qgis-builder:3 test
-
-after_success:
-    - docker run -ti --rm -v $TRAVIS_BUILD_DIR:/io openquake/qgis-builder:3 build_apidoc
-    - docker run -ti --rm -v $TRAVIS_BUILD_DIR:/io openquake/qgis-builder:3 build_manual_html
+    - cd scripts
+    - ./run_unit_tests.sh
```
Comment (daniviga) [inline .travis.yml]: Please add a `FIXME` regarding the possibility to use a multi-stage job

File: .travis.yml [hunk 2]
```diff
@@ -23,19 +18,22 @@ install:
...
+    - docker rm -f qgis
+    - ./run_functional_tests.sh
+    - docker rm -f qgis
+    - ./run_integration_tests.sh
+
+#after_success:
```
Comment (daniviga) [inline .travis.yml]: Please add a `FIXME` we need to restore these tests based on the official QGIS container, but I would address this on a separated PR

File: .travis.yml (hunk 1 above)
Comment (marioba) [reply .travis.yml]: Done

File: .travis.yml (hunk 2 above)
Comment (marioba) [reply .travis.yml]: Done

**Comment by ptormene** [comment]
> Travis currently fails because one of the integration tests is not working.

I've fixed the failing test in the branch `fix-agg_curves-rlzs`

**Pull request merged**